build.js

```diff
@@ -218,74 +237,84 @@ let modelFileIndex = [];
-                const generatedHtmlFile = `${relative}/${fileNameNoExt}.html`;
-                const serverRoot = process.env.SERVER_ROOT;
-                const templateResult = nunjucks.render('model.njk', { serverRoot: serverRoot, modelFile: modelFile, modelVersion: modelVersion, filePath: `${relative}/${fileNameNoExt}`, umlURL: umlURL });
-                modelFileIndex.push({htmlFile: generatedHtmlFile, modelFile: modelFile, modelVersion: modelVersion});
-                console.log(`Processed ${modelFile.getNamespace()} (${modelVersion})`);
-
-                fs.writeFile( `./build/${generatedHtmlFile}`, templateResult, function (err) {
-                    if (err) {
-                        return console.log(err);
+        // skip if this CTO file is not compatible with the current runtime
+        if( isCompatible(concertoVersion, modelText) ) {
+            const modelManager = new ModelManager();
+            if(semver.satisfies(concertoVersion, '0.82.x')) {
+                // load system model if we are using 0.82
+                const systemModel = fs.readFileSync(rootDir + '/cicero/base.cto', 'utf8');
+                modelManager.addModelFile(systemModel, 'base.cto', false, true);
+            }
+            const modelFile  = new ModelFile(modelManager, modelText, file);     
+            console.log(`Processing ${modelFile.getNamespace()}`);
+            let modelFilePlantUML = '';
+            // passed validation, so copy to build dir
+            const dest = file.replace('/src/', '/build/');
+            const destPath = path.dirname(dest);
+            const relative = destPath.slice(buildDir.length);
+            // console.log('dest: ' + dest);
+            // console.log('destPath: ' + destPath);
+            // console.log('relative: ' + relative);
+            
+            const fileName = path.basename(file);
+            const fileNameNoExt = path.parse(fileName).name;
+    
+            await fs.ensureDir(destPath);
+            let umlURL = '';
+            try {
+                // Find the model version
+                const modelVersionStr = relative.match(/v\d+(\.\d+){0,2}/g);
+                const isLegacyModelVersionScheme = modelVersionStr !== null && modelVersionStr.length === 1;
+                if (!isLegacyModelVersionScheme) { // Skip indexing models with the old versioning scheme, they have all been migrated now
+                    const semverStr = modelFile.getName().split("@").pop().slice(0,-4);
+                    const isSemverVersionScheme = semver.valid(semverStr);
+                    const modelVersion = isSemverVersionScheme ? `(${semverStr})` : '(0.1.0)';
+    
+                    modelManager.addModelFile(modelFile, modelFile.getName(), true);
+    
+                    // use the FORCE_PUBLISH flag to disable download of
+                    // external models and model validation
+                    if(!process.env.FORCE_PUBLISH) {
```

Comment:
Is `FORCE_PUBLISH` new? Should it be a parameter of `updateExternalModels`?

Reply:
No, it's not new -- we always had this env var for the script.